### ISSUE TITLE
[0.14 regression] Array initialization produces zeros

### ISSUE THREAD

**Issue by inducer** [pocl/pocl#445]
```
#if __OPENCL_C_VERSION__ < 120
#pragma OPENCL EXTENSION cl_khr_fp64: enable
#endif

__kernel void __attribute__ ((reqd_work_group_size(1, 1, 1))) loopy_kernel(__global double *__restrict__ out)
{
  double tmp[3 * 3] = { 1.3812074215272312, -1.2421753114746372, -1.848378911918023, -0.8717569757582238, 0.7645624900304978, 1.4677250309229288, -0.33230889400340996, 1.14597886763893, -0.9004984264860738 };

  for (int j = 0; j <= 2; ++j)
    for (int i = 0; i <= 2; ++i)
      out[3 * i + j] = tmp[3 * i + j];
}
```

prints

```
[[ 0.  0.  0.]
 [ 0.  0.  0.]
 [ 0.  0.  0.]]
```

on 9ee526a1 with a patched LLVM/Clang 4.0 (master), but nonzero on pocl 0.13 and any other implementation I've tested with.

**Comment by pjaaskel** [comment]
Can you please provide a launcher for this also?

**Comment by pjaaskel** [comment]
OK. I see the problem (in bitcode). It thinks 'double tmp' is an automatic local.

**Comment by pjaaskel** [comment]
I wrote the reproducer host code locally, no need for it anymore.

**Comment by pjaaskel** [comment]
Fixed in 839cdc95894b8

**Comment by inducer** [comment]
Thanks! (And sorry about not responding sooner regarding the host code.)
